tests/test_official.py

```diff
@@ -264,9 +287,222 @@ def check_defaults_type(ptb_param: inspect.Parameter) -> bool:
     return DefaultValue.get_value(ptb_param.default) is None
 
 
+def check_param_type(
+    ptb_param: inspect.Parameter, tg_parameter: list[str], obj: FunctionType | type
+) -> bool:
+    """This function checks whether the type annotation of the parameter is the same as the one
+    specified in the official API. It also checks for some special cases where we accept more types
+
+    Args:
+        ptb_param (inspect.Parameter): The parameter object from our methods/classes
+        tg_parameter (list[str]): The table row corresponding to the parameter from official API.
+        obj (object): The object (method/class) that we are checking.
+
+    Returns:
+        :obj:`bool`: The boolean returned represents whether our parameter's type annotation is the
+        same as Telegram's or not.
+    """
+    # In order to evaluate the type annotation, we need to first have a mapping of the types
+    # specified in the official API to our types. The keys are types in the column of official API.
+    TYPE_MAPPING: dict[str, set[Any]] = {
+        "Integer or String": {int | str},
+        "Integer": {int},
+        "String": {str},
+        r"Boolean|True": {bool},
+        r"Float(?: number)?": {float},
+        r"Array of (?:Array of )?[\w\,\s]*": {Sequence},
+        r"InputFile(?: or String)?": {FileInput},
+    }
+
+    tg_param_type: str = tg_parameter[1]  # Type of parameter as specified in the docs
+    is_class = inspect.isclass(obj)
+    mapped: set[type] = _get_params_base(tg_param_type, TYPE_MAPPING)
+
+    if not mapped:  # no match found, it's from telegram module
+        # it could be a list of objects, so let's check that:
+        objs = _extract_words(tg_param_type)
+        # We want to store both string version of class and the class obj itself. e.g. "InputMedia"
+        # and InputMedia because some annotations might be ForwardRefs.
+        if len(objs) >= 2:  # We have to unionize the objects
+            mapped_type: tuple[Any, ...] = (_unionizer(objs, False), _unionizer(objs, True))
+        else:
+            mapped_type = (
+                getattr(telegram, tg_param_type),  # This will fail if it's not from telegram mod
+                ForwardRef(tg_param_type),
+                tg_param_type,  # for some reason, some annotations are not ForwardRefs, i.e. str.
+            )
+        print("len mapped 0 ", mapped_type)
+    elif len(mapped) == 1:
+        print("len mapped 1 ", mapped)
+        mapped_type = mapped.pop()
```

Comment:
what about `len(mapped) > 1`?

Reply:
should never happen, I can add an assert for that